### PR TITLE
Added disks module

### LINKED ISSUE #842: Add Compute Disk module
## Description

Add module(s) for creating and configuring data disk for Compute VM(s).

Use of `Microsoft.Compute/disks` resource.

### PR DESCRIPTION
**Pull request by Msanzdelrio**
# Change

***Feel free to remove this sample text***
>Thank you for your contribution !

Added disks bicep module. Closes #842 

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

### CODE REVIEW

File: arm/Microsoft.Compute/disks/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  name: '${uniqueString(deployment().name, location)}-AvSet-Rbac-${index}'
+  params: {
+    principalIds: roleAssignment.principalIds
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    resourceId: disk.id
+  }
+}]
+
+@description('The resource group the  disk was deployed into')
+output diskResourceGroup string = resourceGroup().name
```
Comment (eriqua) [inline arm/Microsoft.Compute/disks/deploy.bicep:182]:
```suggestion
output resourceGroupName string = resourceGroup().name
```

File: arm/Microsoft.Compute/disks/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+@description('The resource ID of the disk')
+output diskResourceId string = disk.id
```
Comment (eriqua) [inline arm/Microsoft.Compute/disks/deploy.bicep:185]:
```suggestion
output resourceId string = disk.id
```

File: arm/Microsoft.Compute/disks/deploy.bicep [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+@description('The name of the disk')
+output diskResourceName string = disk.name
```
Comment (eriqua) [inline arm/Microsoft.Compute/disks/deploy.bicep:188]:
```suggestion
output name string = disk.name
```

File: arm/Microsoft.Compute/disks/readme.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+}
+```
+
+## Outputs
+
+| Output Name | Type | Description |
+| :-- | :-- | :-- |
+| `diskResourceGroup` | string | The resource group the  disk was deployed into |
+| `diskResourceId` | string | The resource ID of the disk |
+| `diskResourceName` | string | The name of the disk |
```
Comment (eriqua) [inline arm/Microsoft.Compute/disks/readme.md:89]:
```suggestion
| `resourceGroupName` | string | The resource group the  disk was deployed into |
| `resourceId` | string | The resource ID of the disk |
| `name` | string | The name of the disk |
```

File: arm/Microsoft.Compute/disks/deploy.bicep (hunk 1 above)
Comment (eriqua) [inline arm/Microsoft.Compute/disks/deploy.bicep:182]: In line with the latest output approach as per issue #623 
```suggestion
@description('The resource group the disk was deployed into')
output resourceGroupName string = resourceGroup().name
```

File: arm/Microsoft.Compute/disks/.parameters/min.parameters.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "value": "sxx-az-disk-x-002"
```
Comment (eriqua) [inline arm/Microsoft.Compute/disks/.parameters/min.parameters.json:6]: In line with resource naming ref #796
```suggestion
            "value": "sxx-az-disk-min-001"
```

File: arm/Microsoft.Compute/disks/.parameters/import.parameters.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "value": "sxx-az-disk-x-003"
```
Comment (eriqua) [inline arm/Microsoft.Compute/disks/.parameters/import.parameters.json:6]:
```suggestion
            "value": "sxx-az-disk-import-001"
```

File: arm/Microsoft.Compute/disks/.parameters/image.parameters.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "value": "sxx-az-disk-x-004"
```
Comment (eriqua) [inline arm/Microsoft.Compute/disks/.parameters/image.parameters.json:6]:
```suggestion
            "value": "sxx-az-disk-image-001"
```

File: .github/workflows/ms.compute.disks.yml [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        options:
+          - major
+          - minor
+          - patch
+      customVersion:
+        description: 'Custom version to apply. Used only if higher than latest'
+        required: false
+        default: '0.0.1'
+  push:
+    branches: [main, users/masanzde/disks-module]
```
Comment (eriqua) [inline .github/workflows/ms.compute.disks.yml:25]: Sorry for noticing only just now, please remove `users/masanzde/disks-module`
```suggestion
    branches: 
      - main
```

Comment (Msanzdelrio) [reply .github/workflows/ms.compute.disks.yml:25]: good catch! fixed :)

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**